docs/kore-syntax.md

```diff
@@ -0,0 +1,143 @@
+Syntax of the KORE language
+===========================
+
+This documents describes the syntax of KORE, as admitted by the Haskell Backend.
+
+It is meant to be an easier to access source than the
+[Semantics of K](docs/semantics-of-k.pdf)
+
+Any change in the KORE Parser infrastructure should be reflected here.
+
+Lexical Syntax
+--------------
+
+### Comments
+
+- C-style comments (non-nested)
+
+### Strings
+
+```
+StringLiteral ::= "\"" {Char}* "\""
+
+Char ::= EscapeChar || AsciiChar || PrintableChar
+
+AsciiChar ::= first 128 ASCII characters
+PrintableChar ::= printable according to the C++ iswprint function
+
+EscapeChar
+  ::= ['\"', '\\', '\f', '\n', '\r', '\t']
+    | "\x" {HexDigit}2
+    | "\u" {HexDigit}4
+    | "\U" {HexDigit}8
+
+HexDigit ::= ['0'..'9', 'A'..'F', 'a'..'f']
+```
+
+### KeyWords
+
+```
+Keywords = "module" | "endmodule" | "sort" | "symbol" | "alias" | "axiom"
+```
+
+### Identifiers
+
+```
+IdFirstChar ::= ['A'..'Z', 'a'..'z']
+IdOtherChar ::= ['0'..'9', '\'', '-']
+IdChar ::= IdFirstChar | IdOtherChar
+GenericId ::= IdFirstChar {IdChar}*
+SymbolId ::= {"\"}?GenericId
+ElemVarId ::= GenericId
+SetVarId ::= "@" GenericId
+SortId ::= GenericId
+ModuleNameId ::= GenericId
+```
+
+An identifier cannot be a keyword
+
+Context-Free Syntax
+-------------------
+
+```
+Definition ::= Attributes {Module}+
+
+Attributes ::= "[" {Pattern ","}* "]"
+
+Module ::= "module" ModuleNameId {Sentences}* "endmodule" Attributes
+
+Sentence
+  ::= ImportStatement
+    | SortDefinition   | HookedSortDefinition
+    | SymbolDefinition | HookedSymbolDefinition
+    | AliasDefinition
+    | Axiom
+    | Claim
+
+ImportStatement ::= "import" ModuleNameId Attributes
+
+SortDefinition ::= "sort" SortId SortParameters Attributes
+HookedSortDefinition ::= "hooked-sort" SortId SortParameters Attributes
+
+SymbolDefinition ::=
+    "symbol" SymbolId SortParameters "(" { Sort ","}* "}" ":" Sort Attributes
```

Comment:
Well, in the SDF https://en.wikipedia.org/wiki/Syntax_Definition_Formalism ,  `{ Sort ","}*` is understood as not having a final comma and this notation was created especially to avoid using extra productions when defining lists.

Maybe we should clarify our syntax definition. And we could indeed define lists, if you think it would make things easier to read for newcomers.

Reply:
I've changed everything to BNF and defined enumerations manually.